src/validation.cpp

```diff
@@ -3445,9 +3450,14 @@ void CChainState::ResetBlockFailureFlags(CBlockIndex *pindex) {
     while (pindex != nullptr) {
         if (pindex->nStatus & BLOCK_FAILED_MASK) {
```

Comment:
might need to tweak this line https://github.com/dashpay/dash/pull/6500/files#diff-97c3a52bc5fad452d82670a7fd291800bae20c7bc35bb82686c2c0a4ea7b5b98R3447